ako-infra/ingestion/avi_infra_controller.go

```diff
@@ -84,23 +84,92 @@ func (a *AviControllerInfra) VerifyAviControllerLicense() error {
 	return nil
 }
 
+func (a *AviControllerInfra) checkNSAnnotations() (error, string) {
+	cloudName := ""
+	nsName := utils.GetAKONamespace()
+	nsObj, err := a.cs.CoreV1().Namespaces().Get(context.TODO(), nsName, metav1.GetOptions{})
+	if err != nil {
+		utils.AviLog.Errorf("Failed to GET the %s namespace details due to the following error :%v", nsName, err.Error())
+		return err, cloudName
+	}
+	if nsObj.Annotations != nil {
+		cloudName = nsObj.Annotations["ako.vmware.com/wcp-cloud-name"]
+		if cloudName != "" {
+			utils.AviLog.Infof("Found cloud %s in ns annotation", cloudName)
+		}
+	}
+	return nil, cloudName
+}
+
+func (a *AviControllerInfra) checkVirtualService() (error, string) {
+	cloudName := ""
+	tenants := make(map[string]struct{})
+	err := lib.GetAllTenants(a.AviRestClient, tenants)
+	if err != nil {
+		utils.AviLog.Errorf("Error during GET tenants %v", err)
+		return err, cloudName
+	}
+	createdBy := lib.AKOPrefix + lib.GetClusterName()
+	vsName := lib.GetClusterName() + "--kube-system-kube-apiserver-lb-svc"
+	SetAdminTenant := session.SetTenant(lib.GetAdminTenant())
+	defer SetAdminTenant(a.AviRestClient.AviSession)
+	for tenantName := range tenants {
+		SetTenant := session.SetTenant(tenantName)
+		SetTenant(a.AviRestClient.AviSession)
```

Comment:
we can list VS in all tenants by setting tenantName as *. No need for listing all the tenants and VS in each tenant.